src/cpp/session/modules/rmarkdown/NotebookQueue.cpp

```diff
@@ -668,7 +668,12 @@ class NotebookQueue : boost::noncopyable
    boost::shared_ptr<core::thread::ThreadsafeQueue<std::string> > pInput_;
 };
 
-static boost::shared_ptr<NotebookQueue> s_queue;
+// NOTE: we previously used a shared pointer here but this caused
+// issues with deletion of the static object during program shutdown;
+// since we already manage the lifetime of the queue appropriately
+// we use a raw pointer and let it leak if the user attempts to quit R
+// while the notebook queue is running
+static NotebookQueue* s_queue;
```

Comment:
This should be initialized to `nullptr`.

Reply:
_Technically_ static objects are initialized to their zero value (and so `nullptr` for pointers) but I agree we should do this for clarity.

Reply:
https://i.imgur.com/3wlxtI0.gifv